lib/puppet/application/device.rb

```diff
@@ -256,6 +274,22 @@ def main
               end.join("\n")
             end
             (puts text)
+          elsif options[:apply]
+            # avoid reporting to server
+            Puppet::Transaction::Report.indirection.terminus_class = :yaml
+            Puppet::Resource::Catalog.indirection.cache_class = nil
+
+            require 'puppet/application/apply'
+            begin
+
+              Puppet[:node_terminus] = :plain
+              Puppet[:catalog_terminus] = :compiler
+              Puppet[:catalog_cache_terminus] = nil
+              Puppet[:facts_terminus] = :network_device
```

Comment:
It'd great if you could centralize all of the indirection configuration as there are 3 places in puppet device where this happens, e.g. `app_defaults`, https://github.com/tphoney/puppet/blob/c3cfec90dd30dc7b86ea00e5f36fadb8e1fd4a35/lib/puppet/application/device.rb#L372-L376, and above. Similarly, it would be great to centralize the SSL cert initialization so it's not in the `devices` loop. That said my comment is not blocking this PR, so feel free to merge.

Reply:
@joshcooper - cool thanks for the feedback - TP is out on PTO for the rest of the week - can I just hit merge for him?